server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -1352,7 +1404,13 @@ public static IndexMetaData fromXContent(XContentParser parser) throws IOExcepti
                             }
                         }
                         builder.primaryTerms(list.toArray());
-                    } else {
+                    }
+                    else if (KEY_ALIASES.equals(currentFieldName)) {
+                        while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                            builder.putAlias(AliasMetaData.builder(parser.text()).build());
+                        }
+                    }
+                    else {
```

Comment:
move these `else`s up (lot's of other occurrences in this file)

Reply:
I've removed these changes as they are also for reading back API-serialized XContent. I did correct `else` placement in other places though.